Other/MergeShops/KagesMerge.cs

```diff
@@ -0,0 +1,90 @@
+//cs_include Scripts/CoreBots.cs
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreStory.cs
+//cs_include Scripts/CoreAdvanced.cs
+using Skua.Core.Interfaces;
+using Skua.Core.Models.Items;
+using Skua.Core.Options;
+
+public class KagesMerge
+{
+    public IScriptInterface Bot => IScriptInterface.Instance;
+    public CoreBots Core => CoreBots.Instance;
+    public CoreFarms Farm = new();
+    public CoreStory Story = new();
+    public CoreAdvanced Adv = new();
+    public static CoreAdvanced sAdv = new();
+
+    public List<IOption> Generic = sAdv.MergeOptions;
+    public string[] MultiOptions = { "Generic", "Select" };
+    public string OptionsStorage = sAdv.OptionsStorage;
+    // [Can Change] This should only be changed by the author.
+    //              If true, it will not stop the script if the default case triggers and the user chose to only get mats
+    private bool dontStopMissingIng = false;
+
+    public void ScriptMain(IScriptInterface bot)
+    {
+        Core.BankingBlackList.AddRange(new[] { "Jade Box Trinket", "Jade Box Jewel", "Jade Box Heirloom" });
+        Core.SetOptions();
+        Core.Logger($"{Core.IsMember}");
+
+        BuyAllMerge();
+
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge()
+    {
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("akiba", 355, findIngredients);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Jade Box Trinket":
+                case "Jade Box Jewel":
+                case "Jade Box Heirloom":
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.EquipClass(ClassType.Solo);
+                    Core.RegisterQuests(1593);

```

Comment:
sorry forgot to mention add //QuestName Quest ID above RegisterQuest
always include the quest name when using quest id